Task 1/2019201419王成瑞/double.h

```diff
@@ -0,0 +1,291 @@
+#include <string.h>
+#include <stdint.h>
+
+#ifndef __MY_DOUBLE_H__
+
+#define __MY_DOUBLE_H__
+
+typedef unsigned char       data_t;
+typedef unsigned char*      pointer_t;
+typedef __uint128_t         uint128_t;
+
+#define DOUBLE_SIZE         __SIZEOF_DOUBLE__
+#define EXP_SIZE            (11)
+#define FRAC_SIZE           (52)
+#define DIV_EXTRA_SIZE      (20)
+
+#define BIAS                ((1 << (EXP_SIZE - 1)) - 1)
+#define DOUBLE_EXP_MAX      ((1 << EXP_SIZE) - 2 - BIAS)
+#define DOUBLE_EXP_MIN      (-BIAS)
+
+#define SPECIAL_EXP         (0x7FFu)
+#define DENORMALIZED_EXP    (0u)
+
+#define VL_1                ((uint128_t)1)
+
+#define swap(a, b)                                                      \
+        {                                                               \
+            __typeof__(a) tmp = a;                                      \
```

Comment:
You should know this is a GNU C extension. But it's fine since you already used `__int128`. The problem is that implement `swap()` as a macro is not a good practice. Consider pass two expressions with side effect to it and it will be evaluated more than once. But you can write `__typeof__(a) *pa = &a, *pa = &b` and work with the pointers instead to avoid this problem.